R/transform.R

```diff
@@ -34,23 +42,32 @@
 #' Transforms
 #' @param value For `[<-`, the replacement Transforms to insert
 #' @name Transforms
-#' @aliases transforms transforms<-
+#' @aliases transforms transforms<- 
 NULL
 
 getTransforms <- function (x) {
     var_entity <- entity(x)
     trans <- var_entity@body$view$transform
-
     if (is.null(trans) || length(trans) == 0) {
         return(NULL)
     }
 
-    # get the insertions
-    inserts <- Insertions(data=trans$insertions)
-    # subtype insertions so that Subtotal, Heading, etc. are their rightful selves
-    inserts <- subtypeInsertions(inserts)
+    if (is.null(trans$insertions) || length(trans$insertions) == 0) {
+        inserts <- NULL
+    } else {
+        inserts <- Insertions(data=trans$insertions)
+        # subtype insertions so that Subtotal, Heading, etc. are their rightful 
+        # selves
+        inserts <- subtypeInsertions(inserts)
+    }
+    
+    # if insertions are null return NULL
+    # TODO: when other transforms are implemented, this should check those too.
```

Comment:
I see this TODO here and above, is it worth breaking this check into a function so that it's more obvious in the future where new transform checks should go? That might also help as a kind of list of currently implemented transforms.

Reply:
Aaah, these comments are confusing, the todo should actually be above the checking if the insertions are null. I've moved them around (and spaced them out). The checks here are just checking if there are insertions, and if there aren't any returning `NULL` quickly. The todo is separate from that (and won't replace that when it's implemented...)